.gitlab-ci.yml

```diff
@@ -12,33 +12,28 @@ variables:
   SINGULARITY_EOS_TEMP_SPACK_VERSION: ""
   SINGULARITY_EOS_TEMP_SPACK_DIR: ""
   SINGULARITY_EOS_SPACK_SPEC: "singularity-eos@main+mpi+python+tests%gcc@${SINGULARITY_EOS_GCC_VERSION} ^openmpi@${SINGULARITY_EOS_OPENMPI_VERSION}"
+  COLOR_CYAN: '\033[1;36m'
+  COLOR_PLAIN: '\033[0m'
 
 before_script:
-  - export HOME=${CI_PROJECT_DIR}
   - export SPACK_DISABLE_LOCAL_CONFIG=true
-  - export SPACK_USER_CACHE_PATH=/tmp/spack-local
-  - cd ${CI_PROJECT_DIR}
-  - unset SPACK_ROOT
-  - export SINGULARITY_EOS_SPACK_VERSION=$([ -z "$SINGULARITY_EOS_TEMP_SPACK_VERSION" ] && echo $XCAP_OSS_SPACK_REF || echo $SINGULARITY_EOS_TEMP_SPACK_VERSION)
+  - export SPACK_USER_CACHE_PATH=${SPACK_USER_CACHE_PATH:-/tmp/spack-local}
+  - export DEVEL_SPACK_CHECKOUT=${DEVEL_SPACK_CHECKOUT:-/tmp/spack}
```

Comment:
is it necessary to put the spack in the /tmp folder? If it's there, then we cannot look at the spack since it's always immediately deleted...

Reply:
I/O is usually faster. Setting `DEVEL_SPACK_CHECKOUT` and `SPACK_USER_CACHE_PATH` to something else like scratch will allow you to retain it when needed (e.g. for debugging).

Reply:
If we go this route, we should document the procedure for retaining the output when desired. (Actually we should do that anyway. I think it's not intuitive to a new person on the project how to get to the spack output from a given run.)

Reply:
well I guess if we are going with modify the `.gitlab-ci.yml` when debugging needed route I supposed it's fine